From 5f523f2d6f8fdd443e42543a07d604d85386aa06 Mon Sep 17 00:00:00 2001
From: liuzhensong <liuzhensong@loongson.cn>
Date: Fri, 13 Aug 2021 16:56:56 +0800
Subject: [PATCH 25/30] gas: fix bug. gas/all/gas.exp

  gas/config/tc-loongarch.c
  gas/testsuite/gas/all/gas.exp
---
 gas/config/tc-loongarch.c     | 3 +++
 gas/testsuite/gas/all/gas.exp | 5 ++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/gas/config/tc-loongarch.c b/gas/config/tc-loongarch.c
index bfd83a456c3..eef37d74908 100644
--- a/gas/config/tc-loongarch.c
+++ b/gas/config/tc-loongarch.c
@@ -157,6 +157,9 @@ md_parse_option (int c, const char *arg)
       break;
     case OPTION_IGNORE:
       break;
+    default:
+      ret = 0;
+      break;
     }
   return ret;
 }
diff --git a/gas/testsuite/gas/all/gas.exp b/gas/testsuite/gas/all/gas.exp
index 389634f6165..4825d19460e 100644
--- a/gas/testsuite/gas/all/gas.exp
+++ b/gas/testsuite/gas/all/gas.exp
@@ -68,7 +68,8 @@ if {    ![istarget alpha*-*-*vms*]
      && ![istarget riscv*-*-*]
      && ![istarget rl78-*-*]
      && ![istarget rs6000*-*-aix*]
-     && ![istarget rx-*-*] } then {
+     && ![istarget rx-*-*]
+     && ![istarget loongarch*-*-*] } then {
     gas_test_error "diff1.s" "" "difference of two undefined symbols"
 }
 
@@ -160,9 +161,11 @@ switch -glob $target_triplet {
 	# symbols on relocs.
 	setup_xfail "m68hc1*-*-*" "m6811-*-*" "m6812-*-*" "rl78-*-*"
 	setup_xfail "riscv*-*-*" "rx-*-*" "vax*-*-*" "xgate*-*-*" "z8k-*-*"
+	setup_xfail "loongarch*-*-*"
 	run_dump_test redef2
 	setup_xfail "m68hc1*-*-*" "m6811-*-*" "m6812-*-*" "rl78-*-*"
 	setup_xfail "riscv*-*-*" "rx-*-*" "vax*-*-*" "xgate*-*-*" "z8k-*-*"
+	setup_xfail "loongarch*-*-*"
 	# rs6000-aix disallows redefinition via .comm.
 	if [is_xcoff_format] {
 	    setup_xfail *-*-*
-- 
2.30.1

